Hazelcast fails to compile on Apple M1 Silicon [HZ-800]

**Describe the bug**
When compiling Hazelcast 5.1-SNAPSHOT on an Apple M1 Macbook Pro 2022...
`mvn clean install -DskipTests`

it fails on two extension modules
`hazelcast-jet-grpc`
`hazelcast-jet-protobuf`

with the following log...

```
[ERROR] 1) com.google.protobuf:protoc:exe:osx-aarch_64:3.13.0
[ERROR]
[ERROR]   Try downloading the file manually from the project website.
[ERROR]
[ERROR]   Then, install it using the command:
[ERROR]       mvn install:install-file -DgroupId=com.google.protobuf -DartifactId=protoc -Dversion=3.13.0 -Dclassifier=osx-aarch_64 -Dpackaging=exe -Dfile=/path/to/file
[ERROR]
[ERROR]   Alternatively, if you host your own repository you can deploy the file there:
[ERROR]       mvn deploy:deploy-file -DgroupId=com.google.protobuf -DartifactId=protoc -Dversion=3.13.0 -Dclassifier=osx-aarch_64 -Dpackaging=exe -Dfile=/path/to/file -Durl=[url] -DrepositoryId=[id]
[ERROR]
[ERROR]   Path to dependency:
[ERROR]   	1) com.hazelcast.jet:hazelcast-jet-grpc:jar:5.1-SNAPSHOT
[ERROR]   	2) com.google.protobuf:protoc:exe:osx-aarch_64:3.13.0
[ERROR]
```
**Solution**

Updating the root `pom.xml` to include dependencies that support Arch64.

`com.google.protobuf.protobuf-java` from `3.13.0` to `3.19.1`

`io.grpc.grpc-protobuf` from `1.34.0` to `1.43.0`

--- a/pom.xml
+++ b/pom.xml
@@ -97,7 +97,7 @@
         <avro.version>1.11.0</avro.version>
         <aws.sdk.version>1.12.120</aws.sdk.version>
         <classgraph.version>4.8.138</classgraph.version>
-        <grpc.version>1.34.0</grpc.version>
+        <grpc.version>1.43.0</grpc.version>
         <guava.version>30.1.1-jre</guava.version>
         <hadoop.version>3.3.1</hadoop.version>
         <jackson.version>2.12.1</jackson.version>
@@ -117,7 +117,7 @@
         <parquet.version>1.12.0</parquet.version>
         <picocli.version>4.4.0</picocli.version>
         <prometheus.version>0.14.0</prometheus.version>
-        <protobuf.version>3.13.0</protobuf.version>
+        <protobuf.version>3.19.2</protobuf.version>
         <scala.version>2.12</scala.version>
         <slf4j.version>1.7.32</slf4j.version>
         <spring.version>4.3.0.RELEASE</spring.version>

